tests/integration/adapter/find-all-test.js

```diff
@@ -164,78 +143,63 @@ test('When all records for a type are requested, records that are already loaded
-  );
-  assert.equal(
-    allRecords.objectAt(1).get('name'),
-    'Alex MacCaw',
-    'the second item in the record array is Alex MacCaw'
-  );
-});
-
-test('When all records for a type are requested, records that are created on the client should be added to the record array.', assert => {
-  assert.expect(3);
-
-  store = createStore({
-    adapter: DS.Adapter.extend(),
-    person: Person,
+    allRecords = store.peekAll('person');
+
+    assert.equal(get(allRecords, 'length'), 2, "the record array's length is 2");
+    assert.equal(
+      allRecords.objectAt(0).get('name'),
+      'Jeremy Ashkenas',
+      'the first item in the record array is Jeremy Ashkenas'
+    );
+    assert.equal(
+      allRecords.objectAt(1).get('name'),
+      'Alex MacCaw',
+      'the second item in the record array is Alex MacCaw'
+    );
   });
 
-  allRecords = store.peekAll('person');
+  test('When all records for a type are requested, records that are created on the client should be added to the record array.', assert => {
+    assert.expect(3);
 
-  assert.equal(
-    get(allRecords, 'length'),
-    0,
-    "precond - the record array's length is zero before any records are loaded"
-  );
+    allRecords = store.peekAll('person');
 
-  store.createRecord('person', { name: 'Carsten Nielsen' });
+    assert.equal(
+      get(allRecords, 'length'),
+      0,
+      "precond - the record array's length is zero before any records are loaded"
+    );
 
-  assert.equal(get(allRecords, 'length'), 1, "the record array's length is 1");
-  assert.equal(
-    allRecords.objectAt(0).get('name'),
-    'Carsten Nielsen',
-    'the first item in the record array is Carsten Nielsen'
-  );
-});
+    store.createRecord('person', { name: 'Carsten Nielsen' });
 
-testInDebug('When all records are requested, assert the payload is not blank', assert => {
-  env.registry.register(
-    'adapter:person',
-    DS.Adapter.extend({
-      findAll: () => resolve({}),
-    })
-  );
-
-  assert.expectAssertion(() => {
-    run(() => store.findAll('person'));
-  }, /You made a 'findAll' request for 'person' records, but the adapter's response did not have any data/);
-});
+    assert.equal(get(allRecords, 'length'), 1, "the record array's length is 1");
+    assert.equal(
+      allRecords.objectAt(0).get('name'),
+      'Carsten Nielsen',
+      'the first item in the record array is Carsten Nielsen'
+    );
+  });
 
-test('isUpdating is true while records are fetched', function(assert) {
-  let findAllDeferred = defer();
-  env.registry.register(
-    'adapter:person',
-    DS.Adapter.extend({
-      findAll() {
-        return findAllDeferred.promise;
-      },
+  testInDebug('When all records are requested, assert the payload is not blank', async function(
+    assert
+  ) {
+    let adapter = store.adapterFor('person');
+    adapter.findAll = () => resolve({});
+
+    assert.expectAssertion(() => {
+      run(() => store.findAll('person'));
```

Comment:
we can clean this `run` up in another pass but I believe either `expectAssertion` itself or a similar util is `promise` aware that we can use instead.

Reply:
I'm not sure what's the best approach here, more info at https://github.com/workmanw/ember-qunit-assert-helpers/issues/18.